Specify the  'vpn_gateway' or 'net_gateway' keywords within the split-tunnel routing settings

PeTeeR-mpl opened the pull request:
Hello,
I propose some modifications to the OpenVPN Clone Server functionality to allow pushing classless static routing table entries 
in the form of "network/mask/vpn_gateway" or "network/mask/net_gateway". This is necessary for the OpenVPN Connect client, as it only accepts the 'vpn_gateway' or 'net_gateway' keywords in its split-tunnel routing configuration.
After applying these modifications, static routing table entries can be defined (via Virtual DHCP Server Settings) in the following formats:
a) network_ip/mask/dest_ip — e.g.: 10.0.0.0/255.255.255.0/10.0.0.1
b) network_ip/mask/net_gateway — e.g.: 10.0.0.0/255.255.255.0/net_gateway
c) network_ip/mask/vpn_gateway — e.g.: 10.0.0.0/255.255.255.0/vpn_gateway

The OpenVPN Connect client only accepts formats b) and c), and ignores format a).
The OpenVPN GUI client accepts all formats.
SoftEther and L2TP/IPsec clients only accept format a), and ignore formats b) and c).

I have also modified the DhcpSet command of the vpncmd program. It now displays the current value of the parameter on the prompt line, for example:

Start Point for Distributed Address Band: [10.0.0.101]: _

When you press <Enter>, the current value will be saved. If you type two apostrophes (''), the current value will be reset to an empty or zero value.

chipitsine commented:
@PeTeeR-mpl  if you do not mind, I'll force-push to your branch (no functional change, just more clear git history)

PeTeeR-mpl commented:
> @chipitsine 
Please accept my apologies. I have deleted those files as I mistakenly uploaded them to the wrong folder.

chipitsine commented:
to preserve your authorship, I  need your email (which is missing in your github profile).
if is not hidden on purpose (to hide from some authoritatian nowdays regime(, can you share details?

PeTeeR-mpl committed:
Specify the 'vpn_gateway' or 'net_gateway' keywords within the split-tunnel routing settings

I propose some modifications to the OpenVPN Clone Server functionality to allow pushing classless static routing table entries
in the form of "network/mask/vpn_gateway" or "network/mask/net_gateway". This is necessary for the OpenVPN Connect client, as it only accepts the 'vpn_gateway' or 'net_gateway' keywords in its split-tunnel routing configuration.
After applying these modifications, static routing table entries can be defined (via Virtual DHCP Server Settings) in the following formats:
a) network_ip/mask/dest_ip — e.g.: 10.0.0.0/255.255.255.0/10.0.0.1
b) network_ip/mask/net_gateway — e.g.: 10.0.0.0/255.255.255.0/net_gateway
c) network_ip/mask/vpn_gateway — e.g.: 10.0.0.0/255.255.255.0/vpn_gateway

The OpenVPN Connect client only accepts formats b) and c), and ignores format a).
The OpenVPN GUI client accepts all formats.
SoftEther and L2TP/IPsec clients only accept format a), and ignore formats b) and c).

I have also modified the DhcpSet command of the vpncmd program. It now displays the current value of the parameter on the prompt line, for example:

Start Point for Distributed Address Band: [10.0.0.101]: _

When you press , the current value will be saved. If you type two apostrophes (''), the current value will be reset to an empty or zero value.

chipitsine committed:
Merge remote-tracking branch 'upstream/master'

PeTeeR-mpl committed:
Specify the 'vpn_gateway' or 'net_gateway' keywords within the split-tunnel routing settings I propose some modifications to the OpenVPN Clone Server functionality to allow pushing classless static routing table entries in the form of "network/mask/vpn_gateway" or "network/mask/net_gateway". This is necessary for the OpenVPN Connect client, as it only accepts the 'vpn_gateway' or 'net_gateway' keywords in its split-tunnel routing configuration. After applying these modifications, static routing table entries can be defined (via Virtual DHCP Server Settings) in the following formats: a) network_ip/mask/dest_ip — e.g.: 10.0.0.0/255.255.255.0/10.0.0.1 b) network_ip/mask/net_gateway — e.g.: 10.0.0.0/255.255.255.0/net_gateway c) network_ip/mask/vpn_gateway — e.g.: 10.0.0.0/255.255.255.0/vpn_gateway  The OpenVPN Connect client only accepts formats b) and c), and ignores format a). The OpenVPN GUI client accepts all formats. SoftEther and L2TP/IPsec clients only accept format a), and ignore formats b) and c).  I have also modified the DhcpSet command of the vpncmd program. It now displays the current value of the parameter on the prompt line, for example:  Start Point for Distributed Address Band: [10.0.0.101]: _  When you press , the current value will be saved. If you type two apostrophes (''), the current value will be reset to an empty or zero value.

chipitsine commented:
thank for the fix.

I still hope to find some time to review
